src/Util/TemplateComponentGenerator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
```

Comment:
After we've used this a bunch and finalized it, we should consider removing the `@internal`.

Reply:
agreed - I'm thinking that once we roll out the other pieces to the "new template generator", make all of those components available for use in userland code.